internal/controller/api/v1alpha1/cluster_types.go

```diff
@@ -25,17 +25,86 @@ import (
+	ClusterNetworkCidr       string `json:"cluster_network_cidr,omitempty"`
+	ClusterNetworkHostPrefix int64  `json:"cluster_network_host_prefix,omitempty"`
+	ServiceNetworkCidr       string `json:"service_network_cidr,omitempty"`
+	ApiVip                   string `json:"api_vip,omitempty"`
+	ApiVipDnsName            string `json:"api_vip_dns_name,omitempty"`
+	IngressVip               string `json:"ingres_vip,omitempty"`
+	MachineNetworkCidr       string `json:"machine_network_cidr,omitempty"`
+	SshPublicKey             string `json:"ssh_public_key,omitempty"`
+	VipDhcpAllocation        bool   `json:"vip_dhcp_allocation,omitempty"`
+	HttpProxy                string `json:"http_proxy,omitempty"`
+	HttpsProxy               string `json:"https_proxy,omitempty"`
+	NoProxy                  string `json:"no_proxy,omitempty"`
+	UserManagedNetworking    bool   `json:"user_managed_networking,omitempty"`
+	AdditionalNtpSource      string `json:"additional_ntp_source,omitempty"`
+	InstallConfigOverrides   string `json:"install_config_overrides,omitempty"`
+	// The name of the secret containing the pull secret
+	PullSecretName string `json:"pull_secret_name"`
+}
+
+// +kubebuilder:validation:Enum=insufficient;ready;error;preparing-for-installation;pending-for-input;installing;finalizing;installed;adding-hosts;cancelled;installing-pending-user-action
+type ClusterState string
+
+type ClusterProgressInfo struct {
+
+	// progress info
+	ProgressInfo string `json:"progress_info,omitempty"`
+
+	// Time at which the cluster install progress was last updated.
+	// Format: date-time
+	ProgressUpdatedAt *metav1.Time `json:"progress_updated_at,omitempty"`
+}
+
+type HostNetwork struct {
+
+	// cidr
+	Cidr string `json:"cidr,omitempty"`
 
-	// Foo is an example field of Cluster. Edit Cluster_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// host ids
+	HostIds []string `json:"host_ids"`
 }
 
 // ClusterStatus defines the observed state of Cluster
 type ClusterStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// +optional
+	State ClusterState `json:"state,omitempty"`
+	// +optional
+	StateInfo string `json:"state_info,omitempty"`
+	// +optional
+	HostNetworks []HostNetwork `json:"host_networks,omitempty" gorm:"-"`
+	// +optional
+	InstallStartedAt string `json:"install_started_at,omitempty"`
+	// +optional
+	InstallCompletedAt *metav1.Time `json:"install_completed_at,omitempty"`
+	// +optional
+	Hosts int `json:"hosts,omitempty"`
+	// +optional
+	Progress ClusterProgressInfo `json:"progress,omitempty"`
+	// +optional
+	ValidationsInfo string `json:"validations_info,omitempty"`
+	// +optional
+	ConnectivityMajorityGroups string `json:"connectivity_majority_groups,omitempty"`
+	// +optional
+	UpdatedAt *metav1.Time `json:"updated_at,omitempty"`
+	// +optional
+	CreatedAt *metav1.Time `json:"created_at,omitempty"`
```

Comment:
Should use creationTimestamp property of the object instead

Reply:
Then if i already have it in the metadata do i need it here as well? 
```
apiVersion: adi.io.my.domain/v1alpha1
kind: Cluster
metadata:
...
  creationTimestamp: "2020-12-30T14:28:29Z"
...
spec:
...
status:
...
  created_at: "2020-12-30T14:28:29Z"
```

Reply:
I wrote that we have it already so don't need it here